src/ArraySet/ArraySet.spec.ts

```diff
@@ -0,0 +1,267 @@
+        })
+
+        it("returns an array with duplicates removed (custom equality)", () => {
+            expect(
+                ArraySet.ofArray(
+                    ["A", "D", "d", "A", "E", "f", "F"],
+                    caseInsensitiveEqualityComparer
+                )
+            ).toStrictEqual(["A", "D", "E", "f"])
+        })
+    })
+
+    describe("union", () => {
+        describe("default equality", () => {
+            it("produces a set union", () => {
+                expect(
+                    pipe(
+                        ArraySet.of([1, 2, 3]),
+                        ArraySet.union(ArraySet.of([2, 3, 4]))
+                    )
+                ).toStrictEqual([1, 2, 3, 4])
+            })
+        })
+
+        describe("custom equality", () => {
+            it("produces a set union", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["A", "B", "C", "D"]),
+                        ArraySet.union(
+                            ArraySet.of(["c", "d", "e"]),
+                            caseInsensitiveEqualityComparer
+                        )
+                    )
+                ).toStrictEqual(["A", "B", "C", "D", "e"])
+            })
+        })
+    })
+
+    describe("intersect", () => {
+        describe("default equality", () => {
+            it("produces a set intersection", () => {
+                expect(
+                    pipe(
+                        ArraySet.of([1, 2, 3, 4, 5]),
+                        ArraySet.intersect(ArraySet.of([2, 4, 6]))
+                    )
+                ).toStrictEqual([2, 4])
+            })
+        })
+
+        describe("custom equality", () => {
+            it("produces a set intersection (1)", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["A", "b", "C", "d", "E"]),
+                        ArraySet.intersect(
+                            ArraySet.of(["c", "e", "B", "X"]),
+                            caseInsensitiveEqualityComparer
+                        )
+                    )
+                ).toStrictEqual(["b", "C", "E"])
+            })
+
+            it("produces a set intersection (2)", () => {
+                expect(
+                    pipe(
+                        ArraySet.of(["c", "e", "B", "X", "Y", "Z", "W"]),
+                        ArraySet.intersect(
+                            ArraySet.of(["A", "b", "C", "d", "E"]),
+                            caseInsensitiveEqualityComparer
+                        )
+                    )
+                ).toStrictEqual(["c", "e", "B"])
+            })
+        })
+    })
+
+    describe("except", () => {
+        it("returns empty if the given array is empty", () => {
+            expect(
+                pipe(ArraySet.of([]), ArraySet.except(ArraySet.of([1, 2, 3])))
+            ).toStrictEqual([])
+        })
+
+        it("returns the given array if the excludeThese array is empty", () => {
```

Comment:
`the excludeThese array` typo?

Reply:
ah yea, copy/pasta mistake

Reply:
https://github.com/fp-toolkit/fp-toolkit/pull/67/commits/4de09c6f9d616f7e826aa5189d7b9d552f06ce53